packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -46,36 +46,74 @@ export class ResourceDownloadQueue {
     this._messagesNeedingResourceRetrieval.push(copy);
   }
 
-  public async startQueue(threadId: string, operation: ImageRequest): Promise<void> {
+  public async startQueue(threadId: string, operation: ImageRequest, options?: { singleUrl: string }): Promise<void> {
```

Comment:
Would be nice to have some comment to document the singleUrl part so the Contosos can know that if they don't pass this option in, all previewUrl from a certain image will be downloaded.

Also, should singleUrl be optional as well?
Would `withSingleUrl` a better name? It reads better together `startQueue withSingleUrl` and adds a bit clarity of what this singleUrl is for.

Reply:
This is closed off as private to us. Developers won't be calling it directly. 
We can make the singleUrl optional in the future without any harm since it is not exposed at all.